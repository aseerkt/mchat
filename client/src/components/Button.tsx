import { cva, VariantProps } from 'cva'
import { cn } from '../utils/style'

const buttonVariants = cva('rounded-lg border-2 border-black', {
  variants: {
    variant: {
      primary: 'bg-black text-white',
      secondary: 'bg-white text-black',
    },
    size: {
      sm: 'h-8 px-2',
      lg: 'h-10 px-3',
    },
  },
  defaultVariants: {
    variant: 'primary',
    size: 'lg',
  },
})

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
  VariantProps<typeof buttonVariants>

export const Button = ({ className, variant, size, ...props }: ButtonProps) => {
  return (
    <button
      className={cn(buttonVariants({ variant, size }), className)}
      {...props}
    />
  )
}
