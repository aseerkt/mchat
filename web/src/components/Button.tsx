import { cva, VariantProps } from 'cva'
import { forwardRef } from 'react'
import { cn } from '../utils/style'

const buttonVariants = cva(
  'rounded-lg  border-2 border-black disabled:bg-gray-300',
  {
    variants: {
      variant: {
        primary: 'bg-black text-white',
        secondary: 'bg-white text-black',
      },
      size: {
        sm: 'h-8 px-2',
        lg: 'h-10 px-3',
      },
      color: {
        default: '',
        info: 'border-blue-500',
        error: 'border-red-500',
        success: 'border-green-500',
      },
    },
    defaultVariants: {
      variant: 'primary',
      size: 'lg',
      color: 'default',
    },
    compoundVariants: [
      {
        variant: 'primary',
        color: 'error',
        class: 'bg-red-500',
      },
      {
        variant: 'secondary',
        color: 'error',
        class: 'text-red-500',
      },
      {
        variant: 'primary',
        color: 'success',
        class: 'bg-green-500',
      },
      {
        variant: 'secondary',
        color: 'success',
        class: 'text-green-500 ',
      },
      {
        variant: 'primary',
        color: 'success',
        class: 'bg-blue-500',
      },
      {
        variant: 'secondary',
        color: 'success',
        class: 'text-blue-500 ',
      },
    ],
  },
)

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
  VariantProps<typeof buttonVariants>

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, color, ...props }, ref) => {
    return (
      <button
        ref={ref}
        className={cn(buttonVariants({ variant, size, color }), className)}
        {...props}
      />
    )
  },
)
