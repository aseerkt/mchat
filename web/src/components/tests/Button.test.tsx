import { render } from '@testing-library/react'
import { Button } from '../Button'

describe('<Button /> tests', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<Button>button</Button>)
    expect(baseElement).toMatchSnapshot()
  })
})
