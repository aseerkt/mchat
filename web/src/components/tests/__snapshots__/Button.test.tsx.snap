// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> tests > should match snapshot 1`] = `
<body>
  <div>
    <button
      class="rounded-lg border-2 border-black disabled:bg-gray-300 bg-black text-white h-10 px-3"
    >
      button
    </button>
  </div>
</body>
`;
